### PR TITLE
Database Migration Details: Empty table should have a status saying 'no backup files' 

### PR DESCRIPTION
**Pull request by stuti149**
This PR fixes #

### CODE REVIEW

**Comment by Charles-Gagnon** [comment]
The title of the PR doesn't match the changes being made - was this submitted correctly?

File: extensions/integration-tests/src/onboarding/developers.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ export const developers: string[] = [
 	'lucyzhang929',
 	'saiavishkarsreerama',
 	'smartguest',
+	'stutijain',
```
Comment (Charles-Gagnon) [inline extensions/integration-tests/src/onboarding/developers.ts:24]: This would go in a separate PR typically

**Review by Charles-Gagnon — COMMENTED** [review]
Looks like you have a merge conflict, please fix that when you get a chance.

Could you also make sure to include relevant details of what is being changed when submitting a PR? A description of the changes being made - and then in cases like this where you're adding/changing UI please always include a screenshot of the changes being proposed. 

You may also want to request reviews from relevant team members - most people only get notifications for ones they're assigned as a reviewer on.

File: extensions/integration-tests/src/onboarding/developers.ts (hunk 1 above)
Comment (stuti149) [reply extensions/integration-tests/src/onboarding/developers.ts:24]: sure will do that

**Comment by stuti149** [comment]
image



Above image is result of the changes done. Previously empty Data Migration details table doesn't show anything. Now it shows the status 'no back up files' , in the format shown in figure.

File: extensions/integration-tests/src/onboarding/developers.ts (hunk 1 above)
Comment (stuti149) [reply extensions/integration-tests/src/onboarding/developers.ts:24]: done

File: extensions/sql-migration/src/dialog/migrationCutover/migrationCutoverDialog.ts [hunk 2]
```diff
@@ -259,14 +260,66 @@ export class MigrationCutoverDialog {
...
+					CSSStyles:{
+						'text-align':'center',
+						'font-size':'large',
+						'font-weight':'bold',
+						'width':'300px'
+					}
+				}).component();
+
+				const _emptyTableSubText = view.modelBuilder.text().withProps({
+					value: "If you expected to see results, please check the connection to the sql server instance",
```
Comment (Charles-Gagnon) [inline extensions/sql-migration/src/dialog/migrationCutover/migrationCutoverDialog.ts:287]: 1. As the hygiene check indicates you should use single quotes for strings, not double quotes
2. This should be localized - see other examples in this file (`loc.<var>`)

Same for the other display strings in this file

**Comment by Charles-Gagnon** [comment]
Please ensure you fix all the hygiene issues before submitting this. 

How are you adding commits? Typically we have some pre-commit hooks that are supposed to prevent hygiene failures from making it up. Did you see those when you were committing?

File: extensions/sql-migration/src/dialog/migrationCutover/migrationCutoverDialog.ts [hunk 3]
```diff
@@ -259,14 +260,66 @@ export class MigrationCutoverDialog {
...
+					width: '100px',
+					CSSStyles:{
+						'text-align':'center'
+					}
+
+				}).component();
+
+				const _emptyTableText = view.modelBuilder.text().withProps({
+					value: "No backup files to show",
+					CSSStyles:{
```
Comment (brian-harris) [inline extensions/sql-migration/src/dialog/migrationCutover/migrationCutoverDialog.ts:278]: be sure to localize this string 'No backup files to show' by adding it to a localized constant in strings.ts

File: extensions/sql-migration/src/dialog/migrationCutover/migrationCutoverDialog.ts (hunk 2 above)
Comment (brian-harris) [inline extensions/sql-migration/src/dialog/migrationCutover/migrationCutoverDialog.ts:287]: same here, please add this string to strings.ts as well

File: extensions/sql-migration/src/dialog/migrationCutover/migrationCutoverDialog.ts [hunk 4]
```diff
@@ -259,14 +260,66 @@ export class MigrationCutoverDialog {
...
+						flexFlow: 'column',
+						alignItems: 'center'
+					})
+					.withItems([
+						_emptyTableImage,
+						_emptyTableText,
+						_emptyTableSubText
+					], { CSSStyles: {
+
+					} })
```
Comment (brian-harris) [inline extensions/sql-migration/src/dialog/migrationCutover/migrationCutoverDialog.ts:307]: this can be removed if empty

File: extensions/sql-migration/src/dialog/migrationCutover/migrationCutoverDialog.ts [hunk 5]
```diff
@@ -555,6 +608,15 @@ export class MigrationCutoverDialog {
 				}
 			});
 
+			if(tableData.length === 0)
+			{
```
Comment (brian-harris) [inline extensions/sql-migration/src/dialog/migrationCutover/migrationCutoverDialog.ts:612]: this document needs formatted... i.e. <shift><alt>+'F'

File: extensions/sql-migration/src/dialog/migrationCutover/migrationCutoverDialog.ts [hunk 6]
```diff
@@ -555,6 +608,15 @@ export class MigrationCutoverDialog {
 				}
 			});
 
+			if(tableData.length === 0)
```
Comment (brian-harris) [inline extensions/sql-migration/src/dialog/migrationCutover/migrationCutoverDialog.ts:611]: Can you confirm which conditions apply the empty table message needs to be displayed:
I believe these conditions apply:
* row count is 0
* when backups are accessed from a file share
* "SQL MI" is the migration target

I'm not sure if these apply
* when "SQL VM" is the migration target
* when backups are accessed from a blob container

File: extensions/sql-migration/src/dialog/migrationCutover/migrationCutoverDialog.ts (hunk 2 above)
Comment (stuti149) [reply extensions/sql-migration/src/dialog/migrationCutover/migrationCutoverDialog.ts:287]: Thanks, will fix it.

File: extensions/sql-migration/src/dialog/migrationCutover/migrationCutoverDialog.ts (hunk 6 above)
Comment (stuti149) [reply extensions/sql-migration/src/dialog/migrationCutover/migrationCutoverDialog.ts:611]: I will look into all these conditions.

File: extensions/sql-migration/src/dialog/migrationCutover/migrationCutoverDialog.ts [hunk 7]
```diff
@@ -502,6 +553,13 @@ export class MigrationCutoverDialog {
 				}
 			});
 
+			if (tableData.length === 0) {
+				this._emptyTableFill.updateCssStyles({
+					'display': 'flex'
+				});
+				this._fileTable.height = '50px';
+
+			}
```
Comment (brian-harris) [inline extensions/sql-migration/src/dialog/migrationCutover/migrationCutoverDialog.ts:562]: will the emptyTableFill display be hidden again if tableData length increases

File: extensions/sql-migration/src/dialog/migrationCutover/migrationCutoverDialog.ts [hunk 8]
```diff
@@ -502,6 +553,13 @@ export class MigrationCutoverDialog {
 				}
 			});
 
+			if (tableData.length === 0) {
```
Comment (brian-harris) [inline extensions/sql-migration/src/dialog/migrationCutover/migrationCutoverDialog.ts:556]: This component should only be displayed when the data is expected to be there eventually but isn't.  
I notice the method above that is loading the data checks this._shouldDisplayBackupFileTable() before adding and showing data in the table... this is because the other scenarios won't have the data or expect it.

File: extensions/sql-migration/src/constants/iconPathHelper.ts [hunk 9]
```diff
@@ -128,5 +129,9 @@ export class IconPathHelper {
 			light: context.asAbsolutePath('images/sendFeedback.svg'),
 			dark: context.asAbsolutePath('images/sendFeedback.svg')
 		};
+		IconPathHelper.emptyTable = {
+			light: context.asAbsolutePath('images/emptyTable.svg'),
+			dark: context.asAbsolutePath('images/emptyTable.svg')
```
Comment (Benjin) [inline extensions/sql-migration/src/constants/iconPathHelper.ts:134]: Can you take a screenshot of how this icon shows up on both light and dark themes in ADS?

File: extensions/sql-migration/src/constants/strings.ts [hunk 10]
```diff
@@ -380,6 +380,8 @@ export const DETAILS_COPIED = localize('sql.migration.details.copied', "Details
 export const CANCEL_MIGRATION_CONFIRMATION = localize('sql.cancel.migration.confirmation', "Are you sure you want to cancel this migration?");
 export const YES = localize('sql.migration.yes', "Yes");
 export const NO = localize('sql.migration.no', "No");
+export const EMPTY_TABLE_TEXT = localize('sql.migration.empty.table.text', "No backup files to show");
+export const EMPTY_TABLE_SUBTEXT = localize('sql.migration.emty.table.subtext', "If you expected to see results, please check the connection to the sql server instance");
```
Comment (Benjin) [inline extensions/sql-migration/src/constants/strings.ts:384]: "SQL Server" should be capitalized.  Does this string also refer to other flavors of SQL (Azure, Edge, etc.)?  If so, it should be run by a PM to figure out the correct wording.

Comment (rchlkm) [reply extensions/sql-migration/src/constants/strings.ts:384]: For now, let's update the string to: "If results were expected, reconfirm the connection to the SQL Server Instance."
We want to avoid using the word "please" in our strings.
I will run this by our content writer next week. 

NIT: emty -> empty
